bin/postgres-ha/bootstrap/pre-bootstrap.sh

```diff
@@ -232,7 +232,7 @@ set_pg_user_credentials() {
 set_bootstrap_method() {
     if [[ ! -v PGHA_BOOTSTRAP_METHOD ]]
     then
-        if [[ -n "$(ls -A "${PATRONI_POSTGRESQL_DATA_DIR}")" ]]
+        if [[ -d "${PATRONI_POSTGRESQL_DATA_DIR}" && -n "$(ls -A "${PATRONI_POSTGRESQL_DATA_DIR}")" ]]
         then
             export PGHA_BOOTSTRAP_METHOD="existing_init"
```

Comment:
Does this actually do anything now?

Reply:
The `PGHA_BOOTSTRAP_METHOD` that is set here is used to set the proper bootstrap method for Patroni as defined in the `postgres-ha-bootstrap.yaml` configuration file.  The logic here will specifically apply when a `PGHA_BOOTSTRAP_METHOD` is not explicitly specified in the container, as will be the case when bootstrapping a cluster without first running a bootstrap job.

https://github.com/CrunchyData/crunchy-containers/blob/e90997fbe777be1dc105ce09bc5d2dcc044e68c6/conf/postgres-ha/postgres-ha-bootstrap.yaml#L3

There is also some logic involving this variable in a few other files as well within the `crunchy-postgres-ha` container, e.g. `bootstrap-postgres-ha.sh`, `pre-bootstrap.sh` and `post-bootstrap.sh`:

https://github.com/CrunchyData/crunchy-containers/blob/master/bin/postgres-ha/bootstrap-postgres-ha.sh
https://github.com/CrunchyData/crunchy-containers/blob/master/bin/postgres-ha/bootstrap/pre-bootstrap.sh
https://github.com/CrunchyData/crunchy-containers/blob/master/bin/postgres-ha/bootstrap/post-bootstrap.sh

Reply:
Great.